includes/class-wc-rest-authentication.php

```diff
@@ -66,6 +67,99 @@ protected function is_request_to_rest_api() {
 		return apply_filters( 'woocommerce_rest_is_request_to_rest_api', $woocommerce || $third_party );
 	}
 
+	/**
+	 * Check if this is a request to WCCOM Site REST API.
+	 *
+	 * @return bool
+	 */
+	protected function is_request_to_wccom_site_rest_api() {
+		if ( empty( $_SERVER['REQUEST_URI'] ) ) {
```

Comment:
Can we try and avoid using $_SERVER vars for getting the current url, you can get this with `add_query_arg( array() )`

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/24123/commits/284239ee4485548486390feba1863cad51bed9cf.